maps/github_commits.py

```diff
@@ -0,0 +1,116 @@
+import csv
+import os
+import subprocess
+from pathlib import Path
+from datetime import datetime
+from nomic import AtlasDataset
+import tempfile
+
+# Currently only one github repo can be inputted at a time. Can run repos having over 1 mill commits but it takes time
+
+# Clones repo
+def clone_repo(repo_url, repo_path):
+    if not os.path.exists(repo_path):
+        subprocess.run(['git', 'clone', '--mirror', repo_url, repo_path])
+    else:
+        subprocess.run(['git', '-C', repo_path, 'fetch', '--all'])
+
+# Gets commit messages and metadata along with it
+def fetch_commits_from_local_repo(repo_path):
+    commit_list = []
+    result = subprocess.run(
+        ['git', '-C', repo_path, 'log', '--format=%H;%an;%ae;%ad;%s'],
+        capture_output=True,
+        text=True,
+        encoding='latin1',
+    )
+    commits = result.stdout.strip().split('\n')
+    for i, commit in enumerate(commits, start=1):
+        try:
+            hash, author, email, date, message = commit.split(';', 4)
+            # Formats date so it can be filtered
+            date_obj = datetime.strptime(date, '%a %b %d %H:%M:%S %Y %z')
+            formatted_date = date_obj.strftime('%Y-%m-%d')
+            commit_list.append({
+                'id': i,
+                'hash': hash,
+                'author': author,
+                'email': email,
+                'date': formatted_date,
+                'message': message
+            })
+        except ValueError as e:
+            print(f"ValueError occurred on commit {i}: {e}. Skipping this commit.")
+    return commit_list
+
+# Saves commits to a CSV file
+def save_commits_to_csv(commits, csv_filename):
+    with open(csv_filename, 'w', newline='', encoding='utf-8') as csvfile:
+        fieldnames = ['id', 'hash', 'author', 'email', 'date', 'message']
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        writer.writeheader()
+        for commit in commits:
+            writer.writerow(commit)
+
+# Creates map
+def create_commit_map(csv_filename, map_name):
+    data = []
+    with open(csv_filename, newline='', encoding='utf-8') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            data.append(row)  
+
+    if not data:
+        raise ValueError("No data found in CSV file.")
+
+    dataset = AtlasDataset(
+        map_name,
+        unique_id_field="id",
+    )
+    dataset.add_data(data=data)
+
+    map = dataset.create_index(
+        indexed_field='message',  #Indexs by message
+        topic_model=True,
+        embedding_model='NomicEmbed'
+    )
+
+    return map
+
+if __name__ == "__main__":
+    repo_urls = input("Enter GitHub Repository URLs separated by commas: ").split(',')
+    all_commits = [] #An attempt at allowing multiple github repos, however it doesnt work
+    repo_names = [] #still trying to figure it out - saying that there isnt enough data points (even though there is)
```

Comment:
Nit: remove wip comments from script?
In general, may be worth cleaning up some of the code comments so formatting is consistent